include/openssl/pkcs7.h

```diff
@@ -332,13 +415,48 @@ OPENSSL_EXPORT int PKCS7_type_is_signedAndEnveloped(const PKCS7 *p7);
 // must be NULL and |flags| must be exactly |PKCS7_NOATTR | PKCS7_BINARY |
 // PKCS7_NOCERTS | PKCS7_DETACHED|.
 //
+// The third mode is used for more general signing and does not require the
+// specification of any flags, but does require |sign_cert|, |pkey|, and |data|
+// to be populated. This mode always behaves as if |PKCS7_NOATTR| and
+// |PKCS7_BINARY| are set. It honors the specification (or elision) of
+// |PKCS7_DETACHED|. It does not allow |PKCS7_NOCERTS|. This mode is provided
+// primarily for compatibility with Ruby.
+//
 // Note this function only implements a subset of the corresponding OpenSSL
 // function. It is provided for backwards compatibility only.
 OPENSSL_EXPORT OPENSSL_DEPRECATED PKCS7 *PKCS7_sign(X509 *sign_cert,
                                                     EVP_PKEY *pkey,
                                                     STACK_OF(X509) *certs,
                                                     BIO *data, int flags);
 
+// PKCS7_verify takes in a |p7| with signed ContentInfo and verifies its
+// signature against |certs| or |store|. If |certs| is specified, this function
+// will attempt to verify |p7|'s signature against those certificates' public
+// keys. If |store| is specified, its contents will be treated as certificate
+// authorities (CAs) for establishing trust of any certificates bundled in |p7|.
+//
+// If |p7| is detached, |indata| must contain the data over which |p7|'s
+// signature was computed. If verification succeeds, the verified content is
+// written to |out| and 1 is returned. On error or verification failure, 0 is
+// returned.
+//
+// Flags: If |PKCS7_NOVERIFY |is specified, trust chain validation is skipped.
+// Regardless of whether |PKCS7_NOATTR| is specified, its behavior is enforced.
```

Comment:
No, it's only used in signing, but we fail verification if the caller specifies a signer info with authenticated attributes (because we don't sign them). We should document this behavior, but this section isn't the best place. I'll move it up.